cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/NamespaceHttpHandler.java

```diff
@@ -160,6 +162,8 @@ public void delete(HttpRequest request, HttpResponder responder, @PathParam("nam
     try {
       // Store#deleteNamespace already checks for existence
       NamespaceMeta deletedNamespace = store.deleteNamespace(namespaceId);
+      // Delete Preferences associated with this namespace
+      preferencesWrapper.deleteProperties(namespace);
```

Comment:
The only reason is to at least logically delete everything that the namespace contains before deleting the namespace. I agree its trivial right now. However I could probably imagine a situation in future where we would want to keep a namespace alive until all the entities it contains have been deleted, before finally deleting the namespace itself from the store.